main.tf

```
@@ -100,6 +100,15 @@ resource "aws_security_group" "ecs" {
 resource "aws_ecs_cluster" "default" {
   name = var.name
   tags = var.tags
+
+  capacity_providers = aws_ecs_capacity_provider.default[*].name
+
+  dynamic "default_capacity_provider_strategy" {
+    for_each = aws_ecs_capacity_provider.default[*]
+    content {
+      capacity_provider = default_capacity_provider_strategy.value["name"]
```

I assume this would work just as well and reads just a bit nicer (IMHO):

```suggestion
      capacity_provider = each.value["name"]
```